### PR TITLE
Allow setting more properties (Hackage, etc) when creating an item

### LINKED ISSUE #325: Allow setting Hackage name right in the item creation dialog

### PR DESCRIPTION
**Pull request by neongreen**
fixes #325

### CODE REVIEW

File: back/src/Guide/Api/Types.hs [hunk 1]
```diff
@@ -245,7 +246,7 @@ data TraitSite route = TraitSite
   ,  _createTrait :: route :-
       Summary "Create a new trait in the given item"
       :> Description "Returns the ID of the created trait."
-      :> ErrorResponse 400 "'content' not provided"
+      :> ErrorResponse 400 "'content' can not be empty"
```
Comment (neongreen) [inline back/src/Guide/Api/Types.hs:249]: Would be nice to have some kind of `NotNullText` so that those would be handled automatically.

Comment (willbasky) [reply back/src/Guide/Api/Types.hs:249]: let's open issue.

**Review by willbasky — APPROVED** [review]

File: back/src/Guide/Api/Types.hs (hunk 1 above)
Comment (neongreen) [reply back/src/Guide/Api/Types.hs:249]: I created an entry in https://www.notion.so/aelve/Backend-technical-debt-45a41ab060b7400bb7d0e6547fa34785

**Comment by neongreen** [comment]
@avele rebase and merge when ready

**Pull request merged**